src/Trace.ts

```diff
@@ -111,7 +111,13 @@ export default class Trace {
 			// Combine heads
 			this.head.unshift.apply(this.head, prefix.head);
 			// Combine records
-			this.records.unshift.apply(this.records, prefix.records);
+			if (prefix.records != null) {
+				if (this.records !== null) {
+					this.records.unshift.apply(this.records, prefix.records);
+				} else {
+					this.records = prefix.records;
```

Comment:
This has to be a clone. In the case of multiple forks / joins, we do not want the records to be shared across all branches